shard/connpool.lua

```diff
@@ -112,60 +166,31 @@ local function zone_list(self)
     return names
 end
 
-local function _on_disconnect(self, srv)
-    self:on_disconnect_one(srv)
-
-    -- check zone and pool
-    local d = 0
-    local all_zones = self:zone_list()
-    for _, name in pairs(all_zones) do
-        local alive = self:all(name)
-        if #alive == 0 then
-            self:on_disconnect_zone(name)
-            d = d + 1
-        end
-    end
-    if d == #all_zones then
-        self:on_disconnect()
-    end
-end
-
 local function monitor_fiber(self)
-    fiber.name("monitor")
+    fiber.name("_" .. self.configuration.pool_name .. "_monitor")
```

Comment:
fiber.name throws error, if a name is too long (longer, than 32 bytes). Please, add a {truncate = true} option to a fiber.name() call.